src/useGeolocation.ts

```diff
@@ -54,17 +52,14 @@ const useGeolocation = (options?: PositionOptions): GeoLocationSensorState => {
         speed: event.coords.speed,
         timestamp: event.timestamp,
       });
-    }
-  };
-  const onEventError = (error: IGeolocationPositionError) =>
-    mounted && setState((oldState) => ({ ...oldState, loading: false, error }));
+    };
+    const onEventError = (error: IGeolocationPositionError) =>
+      setState((oldState) => ({ ...oldState, loading: false, error }));
 
-  useEffect(() => {
     navigator.geolocation.getCurrentPosition(onEvent, onEventError, options);
-    watchId = navigator.geolocation.watchPosition(onEvent, onEventError, options);
+    let watchId = navigator.geolocation.watchPosition(onEvent, onEventError, options);
 
     return () => {
-      mounted = false;
       navigator.geolocation.clearWatch(watchId);
     };
   }, []);
```

Comment:
Ideally, options should be added in the deps array but the changes the behavior of the hook